client/src/Widgets/ActivityTable.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { forwardRef, ReactElement } from "react";
+import {
+  Avatar,
+  Text,
+  Icon,
+  Tooltip,
+  Editable,
+  EditableInput,
+  EditablePreview,
+  TableContainer,
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Td,
+  Tbody,
+  Box,
+  HStack,
+  Menu,
+  MenuButton,
+  MenuList,
+} from "@chakra-ui/react";
+import { GoKebabVertical } from "react-icons/go";
+import { FaFolder } from "react-icons/fa";
+import { RiDraftFill } from "react-icons/ri";
+import { MdAssignment } from "react-icons/md";
+import { BsPeopleFill } from "react-icons/bs";
+import { useFetcher, useNavigate } from "react-router-dom";
+import { AssignmentStatus } from "../Tools/_framework/Paths/ActivityEditor";
+
+export default forwardRef(function ActivityTable(
+  {
+    content,
+    suppressAvatar,
+    showOwnerName,
+    showAssignmentStatus,
+    showPublicStatus,
+  }: {
+    content: {
+      cardLink?: string;
+      id: number;
+      assignmentStatus?: AssignmentStatus;
+      isFolder?: boolean;
+      isPublic?: boolean;
+      isShared?: boolean;
+      title: string;
+      ownerName?: string;
+      menuItems?: ReactElement;
+      editableTitle?: boolean;
```

Comment:
Could `editableTitle` and `autoFocusTitle`, be properties of `ActivityTable`? Or are they mixed for a given table?

Reply:
I haven't seen a situation so far where editableTitle varies between rows; if you don't think there are any cases where that might happen, I'll add it as a property of ActivityTable. autoFocusTitle, however, is used to autofocus a folder's title that was just created, and it wouldn't make much sense to have it be true/false for the entire table.